sdks/java/harness/src/test/java/org/apache/beam/fn/harness/control/ProcessBundleHandlerTest.java

```diff
@@ -790,6 +811,367 @@ public void testPTransformStartExceptionsArePropagated() {
     assertThat(handler.bundleProcessorCache.getCachedBundleProcessors().get("1L"), empty());
   }
 
+  private static final class SimpleRecordingDoFn extends DoFn<KV<String, String>, String> {
+    private static final TupleTag<String> MAIN_OUTPUT_TAG = new TupleTag<>("mainOutput");
+    private static final String TIMER_FAMILY_ID = "timer_family";
+
+    @TimerFamily(TIMER_FAMILY_ID)
+    private final TimerSpec timer = TimerSpecs.timerMap(TimeDomain.EVENT_TIME);
+
+    static List<String> consumedData = new ArrayList<>();
```

Comment:
It would be best if you could forward the inputs as outputs and validate your tests that way instead of relying on statics that can cause tests to enter a bad state if things don't go as expected.